gui_classes/GuiSave.py

```diff
@@ -0,0 +1,212 @@
+import functools
+import os
+
+import numpy as np
+from PyQt5 import QtWidgets, QtCore, QtMultimedia, QtGui
+from PyQt5.QtCore import QUrl, QTimer
+from PyQt5.QtGui import QIcon
+
+
+from generated.gui_save_window_generated import Ui_SaveWindow
+
+from sss.dataclasses import SaveWavParams, SaveEvalParams
+from sss.commander import save, save_eval
+
+
+class GUISave(Ui_SaveWindow):
+    def __init__(self, my_data, widget):
+        self.my_data = my_data
+        self.widget = widget
+
+        self.window = None
+        self.main_ui = None
+        self.timer = None
+        self.display_eval_res = False
+        self.dynamicWidgets = []
+        self.player = QtMultimedia.QMediaPlayer()
+        self.mean_eval = {'vocals': {}, 'drums': {}, 'bass': {}, 'other': {}}
+
+    def add_features(self, window, gui_main):
+        self.window = window
+        self.main_ui = gui_main
+
+        self.runButton.setDisabled(True)
+
+        self.runButton.clicked.connect(self.run_handler)
+        self.runButton.setIcon(QIcon('resources\\play_img.png'))
+        self.volumeDownButton.clicked.connect(self.volumeDown_handler)
+        self.volumeDownButton.setIcon(QIcon('resources\\vol_down_img.png'))
+        self.volumeUpButton.clicked.connect(self.volumeUp_handler)
+        self.volumeUpButton.setIcon(QIcon('resources\\vol_up_img.png'))
+        self.returnButton.clicked.connect(self.return_handler)
+
+    def add_save_results(self):
+        self.timer = QTimer()
+
+        font = QtGui.QFont()
+        font.setFamily("Segoe UI")
+        font.setPointSize(12)
+
+        index = 0
+        for index, result in enumerate(self.my_data.audio_wave):
+            labelName = QtWidgets.QLabel(self.centralwidget)
+            labelName.setFont(font)
+            labelName.setText(f'{result[0].value}')
+            labelName.setMinimumSize(QtCore.QSize(120, 40))
+
+            listenButton = QtWidgets.QPushButton(self.centralwidget)
+            listenButton.setText('Listen')
+            listenButton.setFont(font)
+            listenButton.clicked.connect(functools.partial(self.play_handler, index))
+            listenButton.setMinimumSize(QtCore.QSize(120, 40))
+
+            saveOutputButton = QtWidgets.QPushButton(self.centralwidget)
+            saveOutputButton.setText('Save Output')
+            saveOutputButton.setFont(font)
+            saveOutputButton.clicked.connect(functools.partial(self.save_output_handler, index))
+            saveOutputButton.setMinimumSize(QtCore.QSize(120, 40))
+
+            saveEvaluationButton = QtWidgets.QPushButton(self.centralwidget)
+            saveEvaluationButton.setText('Save Evaluation')
```

Comment:
Pokazanie wynikow ewaluacji po najechaniu na ten przycisk jest fajne, ale mam dwie uwagi:
 1. Z jakiego powodu jest to tylko dla ostatniego elementu?
 2. Czy moglo by byc wlaczane po najechaniu na nazwę elementu tak, jak to bylo na makiecie?